test/unit/teacher_set_test.rb

```diff
@@ -182,13 +193,293 @@ class TeacherSetTest < ActiveSupport::TestCase
+      @mintest_mock2.verify
+      assert_equal(188, resp)
+    end
+
+    it 'test StandardError while updating elastic-search' do
+      resp = nil
+      @mintest_mock1.expect(:call, @expected_resp)
+      @mintest_mock2.expect(:update_document_by_id, @es_doc, [nil, @expected_resp])
+    
+      resp = assert_raises(StandardError) do
+        @ts_obj.stub :teacher_set_info, @mintest_mock1 do
+          ElasticSearch.stub :new, @mintest_mock2 do
+            resp = @ts_obj.create_or_update_teacherset_document_in_es
+          end
+        end
+      end
+      assert_equal(ELASTIC_SEARCH_STANDARD_EXCEPTION[:code], resp.code)
+      assert_equal(ELASTIC_SEARCH_STANDARD_EXCEPTION[:msg], resp.message)
+      @mintest_mock1.verify
+    end
   end
 
 
+  # Delete teacher-set document in ES
+  describe "delete teacherset record from elastic search" do
+    it 'test delete_teacherset_record_from_es method' do
+      expected_resp = {"_index" => "teacherset", "_type" => "teacherset", "_id" => @teacher_set.id, "_version" => 2, "result" => "deleted", 
+        "_shards" => {"total" => 2, "successful" => 1, "failed" => 0}, "_seq_no" => 294, "_primary_term" => 2}
+      @mintest_mock1.expect(:delete_document_by_id, expected_resp, [@teacher_set.id])
+      resp = nil
+      ElasticSearch.stub :new, @mintest_mock1 do
+        resp = @teacher_set.delete_teacherset_record_from_es
+      end
+      @mintest_mock1.verify
+      assert_equal(184, resp)
+    end
+
+    it 'test StandardError while deleting elastic-search document' do
+      expected_resp = {"_index" => "teacherset", "_type" => "teacherset", "_id" => @teacher_set.id, "_version" => 2, "result" => "deleted", 
+        "_shards" => {"total" => 2, "successful" => 1, "failed" => 0}, "_seq_no" => 294, "_primary_term" => 2}
+      @mintest_mock1.expect(:delete_document_by_id, expected_resp, [nil])
+      resp = nil
+      resp = assert_raises(StandardError) do
+        ElasticSearch.stub :new, @mintest_mock1 do
+          resp = @teacher_set.delete_teacherset_record_from_es
+        end
+      end
+      assert_equal(ELASTIC_SEARCH_STANDARD_EXCEPTION[:code], resp.code)
+      assert_equal(ELASTIC_SEARCH_STANDARD_EXCEPTION[:msg], resp.message)
+    end
+  end
+
+
+  # test teacher set object method.
+  describe "test teacher set object" do
+    it 'test teacher set object' do
+      resp = @teacher_set.teacher_set_info
+      assert_equal(@teacher_set.title, resp[:title])
+      assert_equal(@teacher_set.bnumber, resp[:bnumber])
+      assert_equal(@teacher_set.subjects.count, resp[:subjects].count)
+    end
+  end
+
+
+  describe 'update set_type value in teacher_set table' do
+    it 'test update_set_type value method' do
+      resp = nil
+      set_type_val = @teacher_set3.set_type
+      @mintest_mock1.expect(:call, set_type_val, [set_type_val])
+      @mintest_mock2.expect(:call, true, [set_type: set_type_val])
+
+      @teacher_set3.stub :derive_set_type, @mintest_mock1 do
```

Comment:
So, for example, here I don't think we _need_ to know that `derive_set_type` was called in a unit test. We only need to know that for given inputs, the right outcomes happened (i.e., the `set_type` was updated. I think that's all that's needed for a unit test.